core/client/fs/src/test/java/alluxio/client/file/BaseFileSystemTest.java

```diff
@@ -571,33 +572,51 @@ public void uriCheckGoodSchemeAuthority() throws Exception {
   public void uriCheckNoSchemeAuthority() throws Exception {
     Configuration.set(PropertyKey.MASTER_HOSTNAME, "localhost");
     Configuration.set(PropertyKey.MASTER_RPC_PORT, "19998");
-    TestAppender appender = new TestAppender();
-    Logger.getRootLogger().addAppender(appender);
-
+    mAppender.track("The URI authority");
+    mAppender.track("The URI scheme");
     AlluxioURI uri = new AlluxioURI("/root");
     mFileSystem.createDirectory(uri);
 
-    if (appender.mEvents.size() > 0) {
-      fail("No events should have been logged.");
-    }
-
-    Logger.getRootLogger().removeAppender(appender);
-    ConfigurationTestUtils.resetConfiguration();
+    assertFalse(mAppender.wasLogged("The URI authority"));
+    assertFalse(mAppender.wasLogged("The URI scheme"));
   }
 
-  private static class TestAppender extends AppenderSkeleton{
+  private static class TestAppender extends AppenderSkeleton {
 
     public List<LoggingEvent> mEvents = new ArrayList<LoggingEvent>();
+    private Hashtable<String, Boolean> mTracked = new Hashtable<>();
+    private Hashtable<String, Boolean> mLogged = new Hashtable<>();
 
     public void close() { }
 
+    /**
+     * Track a specific string to search for in logging events.
+     */
+    public void track(String s) {
+      mTracked.put(s, false);
+    }
+
+    /**
+     * Determines whether a mTracked string was mLogged.
+     */
+    public boolean wasLogged(String s) {
+      return mLogged.containsKey(s);
+    }
+
     public boolean requiresLayout() {
       return false;
     }
 
     @Override
     protected void append(LoggingEvent event) {
       mEvents.add(event);
+
+      for (String key : mTracked.keySet()) {
```

Comment:
this is more efficient than re-scanning in each call to `wasLogged()`, but it also adds more complexity and code overhead in each test that uses it. The lookups will still be sub-millisecond even with the less efficient approach. Normally I'd say leave it as-is, but this will be a model for how we'll test log messages in the future, so I'd like to make it as easy to use as we can